src/content/blog/2023-09-20-tusd-200.md

```diff
@@ -0,0 +1,65 @@
+
+One topic that has been especially tricky in the past are concurrent upload requests. tusd does not support concurrent requests to the same upload resource, as parallel writes could trigger data loss. This rule is enforced through locks, which must be acquired before accessing or modifying an upload. While it solves the problem of concurrent access, it creates another one: When an ongoing upload is interrupted by network issues, the server might be unaware of this interruption and assume that the upload is still continuing and thus holding onto the upload lock. The client, on the other hand, might realize this interruption and abandon the previous upload request to resume the upload. This resumption would fail because the upload is still locked from the previous request, until the connection times out. Depending on the setup, such a timeout can be lengthy and thus cause annoyances for the end-users.
+
+As a solution, tusd v2 allows new requests to ask previous requests to the same upload resource to release their upload lock in a controlled and safe manner. This solves the problem of locked uploads during resumption, while also ensuring no data corruption. You can read more about the topic of upload locks in the [corresponding documentation](https://github.com/tus/tusd/blob/main/docs/locks.md).
+
+## Further improvements
+
+Besides an improved hook system and more resilience, tusd v2 also includes enhancements in the follow areas:
+
+- **Parts are uploaded in parallel to AWS S3:** To implement S3-backed resumable upload, tusd splits an upload into multiple parts and transfers each part individually to S3 using its [multipart uploading functionality](https://docs.aws.amazon.com/AmazonS3/latest/userguide/mpuoverview.html). In tusd v2, the part transfers are now parallelized to achieve higher throughput for the end user.
+- **Graceful shutdown:** Upon receiving a signal to shut down (i.e. SIGINT, SIGTERM or Ctrl+C), tusd will now attempt a graceful exit: First it stops accepting new connections before ending all open uploads while giving the storages enough time to save all buffered data. This reduces data loss and makes reloading tusd more efficient.
+- **Online profiling support:** Powered by Go's [pprof package](https://pkg.go.dev/net/http/pprof), you can now profile the CPU and memory usage of tusd while it is running - even in production with little overhead! This can be enabled using the new `-expose-pprof` option.
+- **Support for the new IETF protocol:** We are working with the HTTP working group of the IETF on a [new standard for resumable file uploads](/blog/2023/08/09/resumable-uploads-ietf). This release includes preliminary support for this experimental protocol via the `-enable-experimental-protocol` flag.
+- **Updated dependencies:** As with every release, we upgraded all dependencies to their latest versions to ensure that we include all available stability and security fixes.
+
+## Breaking Changes
```

Comment:
I thought about putting breaking changes too in tus node blog post but refrained in the end. Figured this is more to get the story and the benefits out and breaking changes lives in releases or migration guides. I don't have a strong opinion on this, but wanted to share the thought.

Reply:
Good point, but I would like to keep them in. I don't plan on writing separate release notes and just want to link to this post instead. So the breaking changes should be in here :)